cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/distributed/AbstractDistributedProgramRunner.java

```diff
@@ -149,6 +152,8 @@ protected AbstractDistributedProgramRunner(TwillRunner twillRunner, LocationFact
     this.cConf = cConf;
     this.eventHandler = createEventHandler(cConf);
     this.secureStoreUpdater = tokenSecureStoreUpdater;
+    this.tmpDir = new File(cConf.get(Constants.CFG_LOCAL_DATA_DIR), cConf.get(Constants.AppFabric.TEMP_DIR));
```

Comment:
Seems like we don't need to have these two as fields, but rather local variables of the run method?